src/test/kotlin/com/exactpro/th2/lwdataprovider/TestSseMessageSearchRequest.kt

```diff
@@ -1,98 +1,164 @@
-        messageSearchReq = SseMessageSearchRequest(grpcRequest)
-        Assertions.assertNotNull(messageSearchReq.startTimestamp)
-        Assertions.assertNotNull(messageSearchReq.endTimestamp)
-        Assertions.assertEquals(TimeRelation.AFTER, messageSearchReq.searchDirection)
+        fun testEndAfterStartDirectionDefault(){
+            val startTimestamp = Timestamp.newBuilder().setNanos(1).build()
+            val endTimestamp = Timestamp.newBuilder().setNanos(2).build()
+            val grpcRequest = MessageSearchRequest.newBuilder().setStartTimestamp(startTimestamp).setEndTimestamp(endTimestamp).build()
+            val messageSearchReq = SseMessageSearchRequest(grpcRequest)
+            Assertions.assertEquals(TimeRelation.AFTER, messageSearchReq.searchDirection)
+            Assertions.assertNotNull(messageSearchReq.startTimestamp)
+            Assertions.assertNotNull(messageSearchReq.endTimestamp)
+            Assertions.assertTrue(messageSearchReq.endTimestamp!! > messageSearchReq.startTimestamp!!)
+        }
 
+        @Test
         // startTimestamp - 1, endTimestamp - 2, searchDirection - BEFORE
-        grpcRequest = MessageSearchRequest.newBuilder().setStartTimestamp(startTimestamp).setEndTimestamp(endTimestamp).
-        setSearchDirection(com.exactpro.th2.dataprovider.grpc.TimeRelation.PREVIOUS).build()
-        assertThrows<InvalidRequestException> {
-            messageSearchReq = SseMessageSearchRequest(grpcRequest)
+        fun testEndAfterStartDirectionBefore(){
+            val startTimestamp = Timestamp.newBuilder().setNanos(1).build()
+            val endTimestamp = Timestamp.newBuilder().setNanos(2).build()
+            val grpcRequest = MessageSearchRequest.newBuilder().setStartTimestamp(startTimestamp).setEndTimestamp(endTimestamp).
+                setSearchDirection(com.exactpro.th2.dataprovider.grpc.TimeRelation.PREVIOUS).build()
+            assertThrows<InvalidRequestException> {
+                SseMessageSearchRequest(grpcRequest)
+            }
         }
 
+        @Test
         // startTimestamp - 3, endTimestamp - 2, searchDirection - BEFORE
-        startTimestamp = Timestamp.newBuilder().setNanos(3).build()
-        grpcRequest = MessageSearchRequest.newBuilder().setStartTimestamp(startTimestamp).setEndTimestamp(endTimestamp).
-        setSearchDirection(com.exactpro.th2.dataprovider.grpc.TimeRelation.PREVIOUS).build()
-        messageSearchReq = SseMessageSearchRequest(grpcRequest)
+        fun testEndBeforeStartDirectionBefore(){
+            val startTimestamp = Timestamp.newBuilder().setNanos(3).build()
+            val endTimestamp = Timestamp.newBuilder().setNanos(2).build()
+            val grpcRequest = MessageSearchRequest.newBuilder().setStartTimestamp(startTimestamp).setEndTimestamp(endTimestamp).
+                setSearchDirection(com.exactpro.th2.dataprovider.grpc.TimeRelation.PREVIOUS).build()
```

Comment:
In kotlin you can use import aliases for the classes. So, instead of using the fully qualified name, you can import it under the alias (e.g. GrpcTimeRelation)